queryfilter/queryfilter.py

```diff
@@ -1,13 +1,57 @@
 from __future__ import absolute_import
 
 from .base import FieldFilter
+from .utils import is_same_class
 
 
 class QueryFilter(object):
+    filter_candidates = {}
+
+    @classmethod
+    def get_filter_key_from_type_and_condition(cls, filter_type,
+                                               filter_condition):
+        return "{}.{}".format(filter_type, filter_condition)
+
+    @classmethod
+    def register_type_condition(cls, filter_type, filter_condition=None):
+        def decorator(filter_class):
+            if not issubclass(filter_class, FieldFilter):
+                raise "Filter to register must be a subclass of FieldFilter."
+            filter_key = cls.get_filter_key_from_type_and_condition(
+                filter_type, filter_condition)
+            if filter_key in cls.filter_candidates and not is_same_class(
+                    cls.filter_candidates[filter_key], filter_class
```

Comment:
Since all members in cls.filter_candidates are expected to be used in the end, it may be just `cls.filter_type_conditions` or `cls.filters_mapping`, or `cls.filters` ... `cls.filter_classes` 🤔

Reply:
hmmm .... I think I'll go with filter_classes 🙂 but it seems not all the members will be access if the front end query only contains certain of them🤔 that's why I use the word candidates haha